src/main.rs

```diff
@@ -206,20 +231,68 @@ fn run(cli: Cli) -> Result {
                 seed_words,
             )
         }
-        Cli::Balance { files, addresses } => {
-            cmd_balance::cmd_balance(collect_addresses(files, addresses)?)
+        Cli::Balance { ledger, files, addresses } => {
+            let addresses = if ledger {
+                ledger_api::get_address()?
+            } else {
+                collect_addresses(files, addresses)?
+            };
+            cmd_balance::cmd_balance(addresses)
         }
         Cli::Hotspots { files, addresses } => {
             cmd_hotspots::cmd_hotspots(collect_addresses(files, addresses)?)
         }
         Cli::Pay {
+            ledger,
+            bones,
             address,
-            amount,
+            mut amount,
             files,
         } => {
-            let pass = get_password(false)?;
-            let wallet = load_wallet(files)?;
-            cmd_pay::cmd_pay(&wallet, &pass, address, amount)
+
+            let amount_in_bones = if bones {
+                amount.parse::<u64>().expect("Bones flag (-b --bones) has been given, but values cannot be parsed as u64. Is this a decimal value?")
+            } else {
+                if let Some(index) = amount.find(".") {
+
+                    let mut digits_after_decimal = amount.len() - index;
+                    // make sure there are not too many decimals
+                    if digits_after_decimal > 8 {
+                        panic!("More than 8 digits exist after the decimal value, which cannot be expressed in Bones!")
+                    }
+
+                    // squash the decimal
+                    let copy = amount.clone();
+                    amount.replace_range(index.., &copy[index+1..]);
+
+                    // make sure there are no more decimals
+                    if let Some(_) = amount.find(".") {
+                        panic!("Multiple decimals in HNT value! Input error")
+                    }
+
+                    while digits_after_decimal <= 8 {
+                        amount.push_str("0");
+                        digits_after_decimal+=1;
+
+                    }
+                } else {
+                    amount.push_str("00000000");
+                };
+                amount.parse::<u64>()?
+            };
+
+            println!("Creating transaction for:");
+            println!("      {:0.*} HNT", 8, (amount_in_bones as f64) / 100000000.0);
+            println!("        =");
```

Comment:
These prints don't really follow the UI model of the rest of the app. are they for development?

Reply:
I wanted to leave some sort of warnings about what the app has understood insofar as HNT and Bones because:

- we are changing the interface (before it was implied Bones, now it's implied HNT)
- it's an easy thing for user to mix up, so giving them a heads up seems reasonable

So then its a question of how to do it? I think these prints are fine. But maybe there's another way? 

There are printed prompts for passwords for the on-host wallets.